navapbc/wic-mt-demo-project-eligibility-screener
# Implement a demo mode

## Ticket

- https://navasage.atlassian.net/wiki/spaces/MWDP/pages/544374811/Readying+the+screener+for+case+study+consumption

## Changes
> What was added, updated, or removed in this PR.

- Add an example CSV output
- Add support for slim <Alert> USWDS component
- Add a `NEXT_PUBLIC_DEMO_MODE` environment variable. When in demo mode:  
  - Display a slim alert across the top of the site
  - Add an alert on the confirmation page
  - Skip submitting data to the Mock API

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

This PR prepares the eligibility screener so that it can be hosted in demo mode as part of the case study publication. Specifically, we don't want to be collecting any user information when in demo mode, so we completely remove connection to the Mock API and add an alert banner.

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

1. Checkout this branch
2. Build and run the container: `docker-compose up --build -d`
3. You should *not* use a docker-compose.yml file that connects this container to the Mock API!
4. Navigate to localhost:3000
5. You should see the alert banner across the top of each page (see screenshot below)
6. Fill out the form wizard
7. You should see the alert on the confirmation page and be able to download the example CSV
8. When you're done, shut down the container: `docker-compose down`

<img width="839" alt="Screen Shot 2022-12-05 at 6 28 23 PM" src="https://user-images.githubusercontent.com/67701/205766851-e264c8cd-2e50-47df-a8cd-6d41aad7004d.png">
<img width="1019" alt="Screen Shot 2022-12-05 at 6 39 39 PM" src="https://user-images.githubusercontent.com/67701/205768845-9d2ee687-8d4c-42be-91bb-129cd939d57e.png">
